### PR TITLE
PTV-1786: Fix delegating type provider inconsistency bug

### PR DESCRIPTION
**Pull request by MrCreosote**
In some cases supplying an absolute type ID could return newer type info than a non-absolute ID, which should be impossible

### CODE REVIEW

**Comment by MrCreosote** [comment]
Travis seems to not be picking up this PR again, although it's running the job from the merge of the last PR

**Comment by MrCreosote** [comment]
There we go

File: src/us/kbase/workspace/test/kbase/DelegatingTypeProviderTest.java [hunk 1]
```diff
@@ -161,6 +165,123 @@ public void getJsonSchemaForAbsoluteType() throws Exception {
...
+				is(new ResolvedType(atype("Foo.Bar-6.2"), "jsonschema here")));
+		
+		// ensure the mocks were called as expected
+		verify(m.ws).getTypeInfo("Foo.Bar-6");
+		verify(m.ws).getTypeInfo("Foo.Bar");
+	}
+	
+	@Test
+	public void jsonSchemaTypeCacheUpatedNoUpdateOnNewerTypes() throws Exception {
+		/* Tests that an type cache isn't updated when requesting an absolute type
```
Comment (ialarmedalien) [inline src/us/kbase/workspace/test/kbase/DelegatingTypeProviderTest.java:210]: `an type cache` fixed locally?

**Review by ialarmedalien — APPROVED** [review]
Minor typo but I'm sure you have it fixed locally already...

File: src/us/kbase/workspace/test/kbase/DelegatingTypeProviderTest.java (hunk 1 above)
Comment (MrCreosote) [reply src/us/kbase/workspace/test/kbase/DelegatingTypeProviderTest.java:210]: <jules>correctamundo</jules>

**Pull request merged**